domain/storageprovisioning/service/service.go

```diff
@@ -66,6 +68,21 @@ type State interface {
 	// GetStorageResourceTagInfoForApplication returns information required to
 	// build resource tags for storage created for the given application.
 	GetStorageResourceTagInfoForApplication(context.Context, application.ID, string) (storageprovisioning.ResourceTagInfo, error)
+
+	// GetStorageIDsForUnit returns the storage IDs for the given unit UUID.
+	//
+	// The following errors may be returned:
+	// - [github.com/juju/juju/domain/application/errors.UnitNotFound] when no
+	// unit exists for the supplied unit UUID.
+	// - [corestorage.InvalidStorageID] when the provided unit UUID is invalid.
+	GetStorageIDsForUnit(ctx context.Context, unitUUID string) ([]corestorage.ID, error)
```

Comment:
we are trying to get away from core. This type we have already identified problems with. Should we just return string here? There is no reason for us to supply anymore meaning to this value other than it acts as an opaque id.